munit/shared/src/main/scala-2/munit/internal/EqualityCompat.scala

```diff
@@ -0,0 +1,13 @@
+package munit.internal
+
+trait EqualityCompat {
+
+  /**
+   * A Scala 2 "polyfill" for the upcoming multiversal equality in Scala 3
+   *
+   * http://dotty.epfl.ch/docs/reference/contextual/multiversal-equality.html
+   */
+  sealed abstract class Eql[A, B]
```

Comment:
cc/ @OlivierBlanvillain @smarter what is the recommended `Eql` "polyfill" for Scala 2 in cross-built library APIs?

Reply:
This is the implementation in the docs

```
package scala
import annotation.implicitNotFound

@implicitNotFound("Values of types ${L} and ${R} cannot be compared with == or !=")
sealed trait Eql[-L, -R]

object Eql {
  object derived extends Eql[Any, Any]
}
```

We can probably copy paste this, and add `implicit` to the `derived` object. We can also probably drop the implicitNotFound annotation for Scala 2 (since it will never be triggered)

Reply:
You're the first person to do it so there's no recommendation yet :). Personally I think I would use separate source files for Scala 2 and 3 in the places where I would want to use Eql.

Reply:
I went with a separate `Compare[A, B]` type-class that I use in both Scala 2 and Scala 3, and in Scala 3 I provide default instances based on `Eql[A, B]`.